internal/services/containerapps/container_app_environment_managed_certificate_resource_test.go

```diff
@@ -0,0 +1,215 @@
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerapps/2024-03-01/managedenvironments"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ContainerAppEnvironmentManagedCertificateResource struct{}
+
+func TestAccContainerAppEnvironmentManagedCertificate_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_app_environment_managed_certificate", "test")
+	r := ContainerAppEnvironmentManagedCertificateResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccContainerAppEnvironmentManagedCertificate_basicUpdateTags(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_app_environment_managed_certificate", "test")
+	r := ContainerAppEnvironmentManagedCertificateResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basicAddTags(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r ContainerAppEnvironmentManagedCertificateResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := managedenvironments.ParseManagedCertificateID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.ContainerApps.ManagedEnvironmentClient.ManagedCertificatesGet(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return pointer.To(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r ContainerAppEnvironmentManagedCertificateResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%[1]s
+
+resource "azurerm_container_app_environment_managed_certificate" "test" {
+  name                         = "acctest-cacert%[2]d"
+  container_app_environment_id = azurerm_container_app_environment.test.id
+
+  subject_name                   = azurerm_container_app_custom_domain.test.name
+  domain_control_validation_type = "CNAME"
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ContainerAppEnvironmentManagedCertificateResource) basicAddTags(data acceptance.TestData) string {
```

Comment:
Can we rename this to
```suggestion
func (r ContainerAppEnvironmentManagedCertificateResource) update(data acceptance.TestData) string {
```